lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3579,6 +3341,97 @@ LogicalResult FIRRTLLowering::visitStmt(SkipOp op) {
   return success();
 }
 
+void FIRRTLLowering::lowerRegConnect(Operation *regOp, Value dest,
+                                     Value srcVal) {
+  // Unroll all accesses since they must be traversed in inverse order.
+  SmallVector<Operation *> accesses;
+  while (dest) {
+    TypeSwitch<Operation *, void>(dest.getDefiningOp())
+        .Case<SubfieldOp, SubindexOp>([&](auto op) {
+          accesses.push_back(op);
+          dest = op.getInput();
+        })
+        .Default([&](auto op) {
+          assert(op == regOp && "invalid lowering");
+          dest = {};
+        });
+  }
```

Comment:
It took me some time to grok that this should always be safe. I was concerned about some wire/node that could legally show up in the `dest`. However, this is definitely impossible.

Reply:
I added a comment to clarify this.